Indicators/CoppockCurve.cs

```diff
@@ -0,0 +1,94 @@
+    /// </summary>
+    public class CoppockCurve : IndicatorBase<IndicatorDataPoint>
+    {
+        private readonly RateOfChangePercent _longRoc;
+        private readonly LinearWeightedMovingAverage _lwma;
+        private readonly RateOfChangePercent _shortRoc;
+
+        public override bool IsReady => _lwma.IsReady;
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="CoppockCurve" /> indicator with its default values.
+        /// </summary>
+        public CoppockCurve()
+            : this("CC_11_14_10", shortRocPeriod: 11, longRocPeriod: 14, lwmaPeriod: 10)
+        {
+        }
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="CoppockCurve"/> indicator
+        /// </summary>
+        /// <param name="shortRocPeriod">The period for the short ROC</param>
+        /// <param name="longRocPeriod">The period for the long ROC</param>
+        /// <param name="lwmaPeriod">The period for the LWMA</param>
+        public CoppockCurve(int shortRocPeriod, int longRocPeriod, int lwmaPeriod) :
+            this($"CC_{shortRocPeriod}_{longRocPeriod}_{lwmaPeriod}", shortRocPeriod, longRocPeriod, lwmaPeriod)
+        {
+        }
+
+        /// <summary>
+        ///     Initializes a new instance of the <see cref="CoppockCurve" /> indicator
+        /// </summary>
+        /// <param name="name">A name for the indicator</param>
+        /// <param name="shortRocPeriod">The period for the short ROC</param>
+        /// <param name="longRocPeriod">The period for the long ROC</param>
+        /// <param name="lwmaPeriod">The period for the LWMA</param>
+        public CoppockCurve(string name, int shortRocPeriod, int longRocPeriod, int lwmaPeriod)
+            : base(name)
+        {
+            _shortRoc = new RateOfChangePercent(shortRocPeriod);
+            _longRoc = new RateOfChangePercent(longRocPeriod);
+            _lwma = new LinearWeightedMovingAverage(lwmaPeriod);
+        }
+
+        /// <summary>
+        /// Resets this indicator to its initial state
+        /// </summary>
+        public override void Reset()
+        {
+            base.Reset();
+            _shortRoc.Reset();
+            _longRoc.Reset();
+            _lwma.Reset();
+        }
+
+        /// <summary>
+        /// Computes the next value of this indicator from the given state
+        /// </summary>
+        /// <param name="input">The input given to the indicator</param>
+        /// <returns>A new value for this indicator</returns>
+        protected override decimal ComputeNextValue(IndicatorDataPoint input)
+        {
+            _shortRoc.Update(input);
+            _longRoc.Update(input);
+            if (_longRoc.IsReady && _shortRoc.IsReady)
+            {
+                var rocSum = _shortRoc + _longRoc;
+                _lwma.Update(input.Time, rocSum);
+                return _lwma;
+            }
+            return decimal.Zero;
```

Comment:
Is there any harm in emitting the non-ready value? When possible, I think it's best that indicators emit an estimate during the non-ready phase vs a zero.

Reply:
Yes, if the `_shortRoc` is ready but the `_longRoc` is not (ROC == 0 if not ready), the LWMA will be updated with the value of the `_shortRoc`. In the default case maybe is not important, but in case the difference between `_shortRoc` and `_longRoc` is big, the `_lwma` (the indicator itself) will be ready with many deceptive values (just the short ROC).

Also, note that the zero value will be returned until both ROC are ready, then it’ll return the `_lwma` value, even when it is not ready.

Reply:
Since this is an oscillating indicator, we're OK with emitting a default value of 0 in the not ready state.